torch_frame/nn/models/excelformer.py

```diff
@@ -37,30 +39,59 @@ def feature_mixup(
         y (Tensor): The target.
         num_classes (int): Number of classes.
         beta (float): The concentration parameter of the Beta distribution.
+            (default: :obj:`0.5`)
+        mixup_type (str): The mixup methods. `none` option means no mixup,
+            options `feature` and `hidden` are `FEAT-MIX` (mixup at feature
+            dimensionality) and `HIDDEN-MIX` (mixup at hidden dimensionality)
+            proposed in ExcelFormer paper. (default: :obj:`none`)
+        mi_scores (Tensor): Mutual information scores only used in the
+            mixup weight calculation for `FEAT-MIX`. (default: :obj:`None`)
 
     Returns:
         x_mixedup (Tensor): The mixedup numerical feature.
         y_mixedup (Tensor): Transformed target of size
             :obj:`[batch_size, num_classes]`
     """
     assert num_classes > 0
+    assert mixup_type in ['none', 'feature', 'hidden']
+
+    beta = torch.tensor(beta, dtype=torch.float32, device=x.device)
     beta_distribution = torch.distributions.beta.Beta(beta, beta)
     shuffle_rates = beta_distribution.sample(torch.Size((len(x), 1)))
-    feat_masks = torch.rand(x.shape, device=x.device) < shuffle_rates
     shuffled_idx = torch.randperm(len(x), device=x.device)
+    assert x.ndim == 3, 'FEAT-MIX or HIDDEN-MIX requires encoded features'
+    if mixup_type == 'feature':
+        assert mi_scores is not None
+        mi_scores = mi_scores.to(x.device)
+        # Hard masks (feature dimension)
+        feat_masks = torch.rand(torch.Size((x.shape[0], x.shape[1])),
```

Comment:
nit: simply write as `torch.rand(x.shape, device=x.device)`

Reply:
As illustrated in the Fig. 3 of ExcelFormer [paper](https://arxiv.org/pdf/2301.02819.pdf), FEAT-MIX & HIDDEN-MIX apply mixup at feature dimension & hidden ones respectively, here the `x` is encoded features with 3 dimensions (batch size, feature amount, hidden size), uniformly writing as `torch.rand(x.shape, device=x.device) < shuffle_rates` in both `feature` and `hidden` options make their behaviors meaningless and different from the paper.

Reply:
I think write as `b, f, d = x.shape` then `torch.rand(torch.Size((b, f)), device=x.device)` for `feature` option (and `torch.rand(torch.Size((b, d)), device=x.device)` for `hidden` option) can also seem readable and brief.

Reply:
Sounds good. Makes sense.